.github/workflows/test-db-changes.yml

```diff
@@ -0,0 +1,61 @@
+name: Test DB Changes
+
+on:
+  push:
+    branches:
+      - main
+      - patch-*
+  pull_request:
+    paths:
+      - 'server/datastore/mysql/schema.sql'
+      - 'server/datastore/mysql/migrations/**.go'
+      - '.github/workflows/test-schema-changes.yml'
+  workflow_dispatch: # Manual
+
+permissions:
+  contents: read
+
+jobs:
+  test-schema-changes:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@84cbf8094393cdc5fe1fe1671ff2647332956b1a # v2
+      with:
+        go-version: '^1.19.1'
+    - name: Checkout Code
+      uses: actions/checkout@629c2de402a417ea7690ca6ce3f33229e27606a5 # v2
+
+    - name: Start Infra Dependencies
+      # Use & to background this
+      run: docker-compose up -d mysql_test &
+
+    - name: Verify test schema changes
+      run: |
+        make dump-test-schema
+        if [[ $(git diff server/datastore/mysql/schema.sql) ]]; then
+          echo "❌ fail: uncommited changes in schema.sql"
+          echo "please run `make dump-test-schema` and commit the changes"
+          exit 1
+        fi
+
+    # TODO: This doesn't cover all scenarios since other PRs might
+    # be merged into `main` after this check has passed.
+    #
+    # We should add a Slack notification or something similar for
+    # when this check fails on `main`.
+    - name: Check migration order
+      run: |
+        all_migrations=($(ls -1 server/datastore/mysql/migrations/tables/20*_*.go | sort -r))
+        new_migrations=($(git diff --name-only origin/main -- server/datastore/mysql/migrations/tables/20*_*.go | sort -r))
```

Comment:
Also, I ran into some issues with this script when I was testing. See https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511

```
Run all_migrations=($(ls -1 server/datastore/mysql/migrations/tables/20*_*.go | sort -r))
[2](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:2)
  all_migrations=($(ls -1 server/datastore/mysql/migrations/tables/20*_*.go | sort -r))
[3](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:3)
  new_migrations=($(git diff --name-only origin/main -- server/datastore/mysql/migrations/tables/20*_*.go | sort -r))
[4](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:4)
  
[5](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:5)
  index=0
[6](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:6)
  for migration in $new_migrations; do
[7](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:7)
    if [ "$migration" != "${all_migrations[$index]}" ]; then
[8](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:8)
      echo "❌ fail: $migration has an older timestamp than ${all_migrations[$index]}"
[9](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:9)
      echo "this might cause problems if this change is merged"
[10](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:10)
      echo "please update the timestamp of $migration"
[11](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:11)
      exit 1
[12](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:12)
    fi
[13](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:13)
    ((index++))
[14](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:14)
  done
[15](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:15)
  shell: /usr/bin/bash -e {0}
[16](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:16)
  
[16](https://github.com/fleetdm/fleet/actions/runs/3070823413/jobs/4960937511#step:6:17)
fatal: bad revision 'origin/main'
```

Reply:
thanks so much for taking the time to test this. It was working but I broke it in a cleanup commit. I will fix and test again.